cli/azd/pkg/azdo/service_connection.go

```diff
@@ -0,0 +1,165 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package azdo
+
+import (
+	"context"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/environment"
+	"github.com/azure/azure-dev/cli/azd/pkg/input"
+	"github.com/azure/azure-dev/cli/azd/pkg/output"
+	"github.com/microsoft/azure-devops-go-api/azuredevops"
+	"github.com/microsoft/azure-devops-go-api/azuredevops/build"
+	"github.com/microsoft/azure-devops-go-api/azuredevops/serviceendpoint"
+)
+
+// authorize a service connection to be used in all pipelines
+func authorizeServiceConnectionToAllPipelines(
```

Comment:
nit: Consider grouping private (lowercased) funcs at the bottom of the file and public (CamelCased) funcs at the top.

Reply:
I kept the private funcs near the exported funcs where they are used for readability.

Reply:
Makes it easer for development but for a unknown returning dev having the funcs grouped by visibility may be more beneficial.